doctr/models/recognition/parseq/pytorch.py

```diff
@@ -0,0 +1,493 @@
+
+from ...classification import vit_s
+from ...utils.pytorch import load_pretrained_params
+from .base import _PARSeq, _PARSeqPostProcessor
+
+__all__ = ["PARSeq", "parseq"]
+
+default_cfgs: Dict[str, Dict[str, Any]] = {
+    "parseq": {
+        "mean": (0.694, 0.695, 0.693),
+        "std": (0.299, 0.296, 0.301),
+        "input_shape": (3, 32, 128),
+        "vocab": VOCABS["french"],
+        "url": None,
+    },
+}
+
+
+class CharEmbedding(nn.Module):
+    """Implements the character embedding module
+
+    Args:
+        vocab_size: size of the vocabulary
+        d_model: dimension of the model
+    """
+
+    def __init__(self, vocab_size: int, d_model: int):
+        super().__init__()
+        self.embedding = nn.Embedding(vocab_size, d_model)
+        self.d_model = d_model
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return math.sqrt(self.d_model) * self.embedding(x)
+
+
+class PARSeqDecoder(nn.Module):
+    """Implements decoder module of the PARSeq model
+
+    Args:
+        d_model: dimension of the model
+        num_heads: number of attention heads
+        ffd: dimension of the feed forward layer
+        ffd_ratio: depth multiplier for the feed forward layer
+        dropout: dropout rate
+    """
+
+    def __init__(
+        self,
+        d_model: int,
+        num_heads: int = 12,
+        ffd: int = 2048,
+        ffd_ratio: int = 4,
+        dropout: float = 0.1,
+    ):
+        super().__init__()
+        self.attention = MultiHeadAttention(num_heads, d_model, dropout=dropout)
+        self.cross_attention = MultiHeadAttention(num_heads, d_model, dropout=dropout)
+        self.position_feed_forward = PositionwiseFeedForward(d_model, ffd * ffd_ratio, dropout, nn.GELU())
+
+        self.attention_norm = nn.LayerNorm(d_model, eps=1e-5)
+        self.cross_attention_norm = nn.LayerNorm(d_model, eps=1e-5)
+        self.query_norm = nn.LayerNorm(d_model, eps=1e-5)
+        self.content_norm = nn.LayerNorm(d_model, eps=1e-5)
+        self.feed_forward_norm = nn.LayerNorm(d_model, eps=1e-5)
+        self.output_norm = nn.LayerNorm(d_model, eps=1e-5)
+        self.attention_dropout = nn.Dropout(dropout)
+        self.cross_attention_dropout = nn.Dropout(dropout)
+        self.feed_forward_dropout = nn.Dropout(dropout)
+
+    def forward(
+        self,
+        target,
+        content,
+        memory,
+        target_mask: Optional[torch.Tensor] = None,
+    ):
+        query_norm = self.query_norm(target)
+        content_norm = self.content_norm(content)
+        target = target.clone() + self.attention_dropout(
+            self.attention(query_norm, content_norm, content_norm, mask=target_mask)
+        )
+        target = target.clone() + self.cross_attention_dropout(
+            self.cross_attention(self.query_norm(target), memory, memory)
+        )
+        target = target.clone() + self.feed_forward_dropout(self.position_feed_forward(self.feed_forward_norm(target)))
```

Comment:
Why are there `clone` calls here ?

Reply:
Pytorch does not allow overriding inplace because it would raise problems in CUDA :) 

The other Option would be to use 2 variables but i personally like the clone() 
way -> minimizing to code a few lines 😅

Reply:
You can do àn only clone one before line 97 and remove the other clones. I have tried before it worked. To verify to be sûre